packages/cozy-harvest-lib/src/services/budget-insight.js

```diff
@@ -166,10 +170,24 @@ export const createOrUpdateBIConnection = async ({
       }
     }
     logger.info('Creating or updating connection...')
-    const connection = await (connId
+    const isUpdate = Boolean(connId)
+    const connection = await (isUpdate
       ? updateBIConnection(config, connId, credsToSend, tempToken)
       : createBIConnection(config, credsToSend, tempToken))
 
+    if (!isUpdate) {
+      const sameAccount = await fetchAccountWithBiConnection({
```

Comment:
actually, we need only a boolean has return type.

Reply:
Since, as you ask in the following comment, it would be better to add the corresponding account id. I prefer to keep it like this.